lib/Window.js

```diff
@@ -23,105 +23,204 @@ module.exports = (Item) => {
     }
 
     acceptClick (click) {
-      assert.ok(click.mouseButton === 0 || click.mouseButton === 1)
+      const { mode, slot, mouseButton } = click
+      assert.ok(
+        (mode <= 6 && mode >= 0) &&
+        (mouseButton >= 0 && mouseButton <= 8) &&
+        ((slot >= 0 && slot < this.inventoryEnd) || slot === -999),
+        'invalid operation')
+
+      switch (click.mode) {
+        case 0:
+          assert.ok(mouseButton <= 1, 'invalid operation')
+          this.mouseClick(click)
+          break
+
+        case 1:
+          assert.ok(mouseButton <= 1, 'invalid operation')
+          this.shiftClick(click)
+          break
+
+        case 2:
+          assert.ok(mouseButton <= 8, 'invalid operation')
+          this.numberClick(click)
+          break
+
+        case 3:
+          assert.ok(mouseButton === 2, 'invalid operation')
+          this.middleClick(click)
+          break
+
+        case 4:
+          assert.ok(mouseButton <= 1, 'invalid operation')
+          this.dropClick(click)
+          break
+
+        case 5:
+          this.dragClick(click)
```

Comment:
if you're doing asserts on the rest, might as well assert that this is in [1,5,9,2,6,10]